File: functions/src/domain/repositories/transaction_repository.ts
import { Transaction } from '../entities/transaction';

export interface TransactionRepository {
  createTransaction(transactionData: Omit<Transaction, 'id'>): Promise<Transaction>;
  getTransaction(id: string): Promise<Transaction | null>;
  getAllTransactions(): Promise<Transaction[]>;
  getTransactionsByYear(year: number): Promise<Transaction[]>;
  updateTransaction(id: string, transactionData: Partial<Transaction>): Promise<Transaction>;
  deleteTransaction(id: string): Promise<void>;
}

File: lib/features/transactions/domain/repositories/transaction_repository.dart
import '../entities/transaction.dart';
import '../entities/annual_growth.dart';

abstract class TransactionRepository {
  Future<List<Transaction>> getTransactions();
  Future<Transaction> getTransaction(String id);
  Future<void> createTransaction(Transaction transaction);
  Future<void> updateTransaction(Transaction transaction);
  Future<void> deleteTransaction(String id);
  Future<AnnualGrowth> getAnnualGrowth(int year);
}
